test/models/tts_test.dart

```diff
@@ -0,0 +1,54 @@
+import 'package:flutter/services.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/mockito.dart';
+import 'package:rtchat/tts_plugin.dart';
+
+void main() {
+  group('TTSQueue', () {
+    late TTSQueue ttsQueue;
+    late MockMethodChannel mockChannel;
+
+    setUp(() {
+      mockChannel = MockMethodChannel();
+      ttsQueue = TTSQueue(mockChannel);
+    });
+
+    test('Queue starts empty', () {
+      expect(ttsQueue.isEmpty, isTrue);
+    });
+
+    test('Speak adds elements to the queue', () {
+      ttsQueue.speak('1', 'First message');
+      ttsQueue.speak('2', 'Second message');
+      expect(ttsQueue.isEmpty, isFalse);
+      expect(ttsQueue.length, equals(2));
```

Comment:
Shouldn't this be 1 because the first invocation gets sent to the channel?

Reply:
Let me know if the changes resolved it. The only way I could figure out how to get it to only have 1 in the queue is to actually "speak" the message by using speakNext(). If this is not what needed to be done let me know!